tests/usr.bin/cc/t_ubsan_int_add_overflow.sh

```diff
@@ -0,0 +1,200 @@
+# Copyright (c) 2018 The NetBSD Foundation, Inc.
+# All rights reserved.
+#
+# This code is derived from software contributed to The NetBSD Foundation
+# by Harry Pantazis.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE NETBSD FOUNDATION, INC. AND CONTRIBUTORS
+# ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+# TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR CONTRIBUTORS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+
+test_target()
+{
+	SUPPORT='n'
+	if ! echo __GNUC__ | cc -E - | grep -q __GNUC__; then 
+		SUPPORT='y'
+	fi
+
+	if ! echo __clang__ | cc -E - | grep -q __clang__; then 
+		SUPPORT='y'
+	fi
+}
+
+atf_test_case int_add_overflow
+int_add_overflow_head() {
+	atf_set "descr" "Test Undefined Behavior for int addition overflows"
+	atf_set "require.progs" "cc"
+}
+
+atf_test_case int_add_overflow_profile
+int_add_overflow_profile_head() {
+	atf_set "descr" "Test Undefined Behavior for int addition overflows with profiling option"
+	atf_set "require.progs" "cc"
+}
+atf_test_case int_add_overflow_pic
+int_add_overflow_pic_head() {
+	atf_set "descr" "Test Undefined Behavior for int addition overflows with position independent code (PIC) flag"
+	atf_set "require.progs" "cc"
+}
+atf_test_case int_add_overflow_pie
+int_add_overflow_pie_head() {
+	atf_set "descr" "Test Undefined Behavior for int addition overflows with position independent execution (PIE) flag"
+	atf_set "require.progs" "cc"
+}
+atf_test_case int_add_overflow32
+int_add_overflow32_head() {
+	atf_set "descr" "Test Undefined Behavior for int addition overflows in NetBSD_32 emulation"
+	atf_set "require.progs" "cc file diff cat"
+}
+
+
+int_add_overflow_body(){
+	cat > test.c << EOF
+#include "stdint.h"
```

Comment:
Why `"` in includes? Code like this is broken. We need to use `<stdint.h>`.
We want to include `<limits.h>` instead of `<stdint.h>` for `INT_MAX`.

Reply:
Where does `int-add-overflow` come from? It's a broken check for me on GCC.

```
$ sudo ./t_ubsan_int_add_overflow int_add_overflow 
t_ubsan_int_add_overflow: WARNING: Running test cases without atf-run(1) is unsupported
t_ubsan_int_add_overflow: WARNING: No isolation nor timeout control is being applied; you may get unexpected failures; see atf-test-case(4)
Executing command [ ./test ]
Fail: incorrect exit status: 0, expected: anything else
stdout:
CHECK

stderr:
test.c:4:53: runtime error: signed integer overflow: 2147483647 + 1 cannot be represented in type 'int'

failed: atf-check failed; see the output of the test for details
```

We need to check for `signed integer overflow`.